samples/Common/cpp/CameraHelper_cppwinrt.cpp

```diff
@@ -0,0 +1,279 @@
+    }
+    catch (...)
+    {
+        if (instance != nullptr)
+        {
+            delete instance;
+        }
+        throw;
+    }
+
+    return instance;
+}
+
+//
+// Initialize camera pipeline resources and register a callback for when new VideoFrames become available.
+//
+void CameraHelper::Initialize()
+{
+    // Initialize MediaCapture with default settings in video-only streaming mode.
+    // We first try to aquire exclusive sharing mode and if we fail, we then attempt again in shared mode
+    // so that multiple instances can access the camera concurrently
+    m_mediaCapture = MediaCapture();
+    auto mediaCaptureInitializationSettings = MediaCaptureInitializationSettings();
+    mediaCaptureInitializationSettings.SharingMode(m_sharingMode);
+    mediaCaptureInitializationSettings.StreamingCaptureMode(StreamingCaptureMode::Video);
+
+    // Register a callback in case MediaCapture fails. This can happen for example if another app is using the camera and we can't get ExclusiveControl
+    m_mediaCapture.Failed({ this, &CameraHelper::MediaCapture_Failed });
+
+    m_mediaCapture.InitializeAsync(mediaCaptureInitializationSettings).get();
+
+    // Get a list of available Frame source and iterate through them to find a video preview or 
+    // a video record source with color images (and not IR, depth or other types)
+    auto frameSourceIterator = m_mediaCapture.FrameSources().First();
+    const wchar_t* cameraName;
+    do
+    {
+        auto mediaStreamType = frameSourceIterator.Current().Value().Info().MediaStreamType();
+        auto sourceKind = frameSourceIterator.Current().Value().Info().SourceKind();
+        cameraName = frameSourceIterator.Current().Value().Info().DeviceInformation().Name().c_str();
+
+        std::wcout << cameraName << L" | MediaStreamType:" << (int)mediaStreamType << L" MediaFrameSourceKind:" << (int)sourceKind << std::endl;
+        if ((mediaStreamType == MediaStreamType::VideoPreview) && (sourceKind == MediaFrameSourceKind::Color))
+        {
+            break;
+        }
+    } while (frameSourceIterator.MoveNext());
+
+    if (!frameSourceIterator.HasCurrent())
+    {
+        frameSourceIterator = m_mediaCapture.FrameSources().First();
+        do
+        {
+            auto mediaStreamType = frameSourceIterator.Current().Value().Info().MediaStreamType();
+            auto sourceKind = frameSourceIterator.Current().Value().Info().SourceKind();
+            cameraName = frameSourceIterator.Current().Value().Info().DeviceInformation().Name().c_str();
+
+            if ((mediaStreamType == MediaStreamType::VideoRecord) && (sourceKind == MediaFrameSourceKind::Color))
+            {
+                break;
+            }
+        } while (frameSourceIterator.MoveNext());
+    }
+    if (!frameSourceIterator.HasCurrent())
+    {
+        std::cerr << "No valid video frame sources were found with source type color.";
+        throw new winrt::hresult(MF_E_INVALIDMEDIATYPE);
```

Comment:
nit: consider [winrt::throw_hresult()](https://docs.microsoft.com/en-us/uwp/cpp-ref-for-winrt/error-handling/throw-hresult)